packages/react/src/shared/types.ts

```diff
@@ -0,0 +1,27 @@
+import { QueryClient } from '@tanstack/react-query';
+import { AnyRouter, MaybePromise } from '@trpc/server';
+
+/**
+ * @internal
+ */
+export interface UseMutationOverride {
+  onSuccess: (opts: {
+    /**
+     * Calls the original function that was defined in the query's `onSuccess` option
+     */
+    originalFn: () => MaybePromise<unknown>;
+    queryClient: QueryClient;
+  }) => MaybePromise<unknown>;
```

Comment:
allow overriding `onError`, `onMutate` etc too?

Reply:
Meh. I don't use it